src/components/SiteInfoInputForm.js

```diff
@@ -0,0 +1,786 @@
+
+    onClickIAmHere = () => {
+        const {siteFormType} = this.props;
+
+        this.props.promptForLocationServicesPermission();
+    };
+
+    onClickSubmit = () => {
+        const {siteFormType, navigate, goBack} = this.props;
+
+        const newSite = {
+            title: this.props.siteTitleText,
+            description: this.props.siteDescriptionText,
+            directions: this.props.siteDirectionsText,
+            alternateSites: this.props.siteAlternateSitesText,
+            nearestTown: this.props.siteNearestTownText,
+            accessibility: this.props.accessibilityOption,
+            facilities: this.props.siteDetailCheckboxesKeys.facilities,
+            features: this.props.siteDetailCheckboxesKeys.features,
+            price: this.props.priceOption,
+            coordinate: {
+                longitude: this.props.readyLongitude,
+                latitude: this.props.readyLatitude
+            },
+            siteImageData: this.props.siteImageData,
+            cellProvider: this.props.cellProviderOption,
+            cellStrength: this.props.cellStrengthOption,
+            county: this.props.countyOption,
+            forest: this.props.forestOption,
+            mvum: this.props.mvumOption,
+        };
+
+        this.props.attemptToUploadSite(newSite, {navigate, goBack}, siteFormType);
+    };
+
+    renderSubmitOptions = () => {
+        const {siteReadyForUpload, siteFormType, navigate} = this.props;
+        const {submitButtonStyle, lastElementStyle, adminOptionsButtonContainerStyle, iconButtonStyle, approveButtonStyle, cancelButtonStyle} = styles;
+
+        if (siteReadyForUpload && siteFormType === site_form_type.ADD) {
+            return (
+                <LargeButton
+                    title={submit}
+                    iconType={'font-awesome'}
+                    iconName={'plus'}
+                    iconColor={'white'}
+                    buttonStyleOverride={submitButtonStyle}
+                    onPress={this.onClickSubmit}
+                />
+            );
+        } else if (siteReadyForUpload && siteFormType === site_form_type.EDIT) {
+            return (
+                <View style={[lastElementStyle, adminOptionsButtonContainerStyle]}>
+
+                    <SmallButton
+                        title={'Cancel'}
+                        iconType={'ionicon'}
+                        iconName={'md-close-circle'}
+                        iconColor={'white'}
+                        buttonStyleOverride={cancelButtonStyle}
+                        onPress={() => navigate(navKeys.SITE_DETAIL)}
+                    />
+
+                    <SmallButton
+                        title={'Update'}
+                        iconType={'ionicon'}
+                        iconName={'md-checkmark-circle'}
+                        iconColor={'white'}
+                        buttonStyleOverride={approveButtonStyle}
+                        onPress={this.onClickSubmit}
+                    />
```

Comment:
If it's an Add screen, show the submit button. If it's an edit screen, show the update or cancel buttons.

Reply:
Note this also uses the new `SmallButton` reusable component.